cumulus_library_covid/covid_symptom/table_pcr.sql

```diff
@@ -24,10 +24,10 @@ WITH obs_interpret AS (
 
 SELECT DISTINCT
     obs_interpret.display AS covid_pcr_result_display,
-    o.lab_code AS covid_pcr_code,
-    o.lab_date AS covid_pcr_date,
-    o.lab_week AS covid_pcr_week,
-    o.lab_month AS covid_pcr_month,
+    o.observation_code AS covid_pcr_code,
+    o.effectivedatetime_day AS covid_pcr_date,
+    o.effectivedatetime_week AS covid_pcr_week,
+    o.effectivedatetime_month AS covid_pcr_month,
```

Comment:
Sure - but I want the simplification promise to go even further :smile: - give me just `effective_day` and that will be the day for whichever of the `effectiveX` fields happened to be filled out. That's right in `core`'s wheelhouse

Reply:
Observation.effective especially triggers me because it offers two fields that are so very similar: effectiveDateTime and effectiveInstant - the only difference being whether the field can hold sub-seconds. But now all consumers have to check both fields.

Reply:
I like the effective day promise - like, i'm fine with saying that should always resolve to a reasonable thing.

week and month are trickier, if only because they are very common binning units for studies - we're basically obviating away date casting downstream at the cost of space on disk. I'm sort of ok with this.

Reply:
for posterity - after clarifying some stuff off thread, we created an issue in core to simplify this kind of rat king type handling https://github.com/smart-on-fhir/cumulus-library/issues/205